src/Client/CompatShim/ShimDurableTaskClient.cs

```diff
@@ -0,0 +1,261 @@
+        };
+
+        string? serializedInput = this.DataConverter.Serialize(input);
+        TaskMessage message = new()
+        {
+            OrchestrationInstance = instance,
+            Event = new ExecutionStartedEvent(-1, serializedInput)
+            {
+                Name = orchestratorName.Name,
+                Version = orchestratorName.Version,
+                OrchestrationInstance = instance,
+                ScheduledStartTime = options?.StartAt?.UtcDateTime,
+            },
+        };
+
+        await this.Client.CreateTaskOrchestrationAsync(message);
+        return instanceId;
+    }
+
+    /// <inheritdoc/>
+    public override Task SuspendInstanceAsync(
+        string instanceId, string? reason = null, CancellationToken cancellation = default)
+        => this.SendInstanceMessageAsync(instanceId, new ExecutionSuspendedEvent(-1, reason), cancellation);
+
+    /// <inheritdoc/>
+    public override Task ResumeInstanceAsync(
+        string instanceId, string? reason = null, CancellationToken cancellation = default)
+        => this.SendInstanceMessageAsync(instanceId, new ExecutionResumedEvent(-1, reason), cancellation);
+
+    /// <inheritdoc/>
+    public override Task TerminateAsync(
+        string instanceId, object? output = null, CancellationToken cancellation = default)
+    {
+        Check.NotNullOrEmpty(instanceId);
+        cancellation.ThrowIfCancellationRequested();
+        string? reason = this.DataConverter.Serialize(output);
+        return this.Client.ForceTerminateTaskOrchestrationAsync(instanceId, reason);
+    }
+
+    /// <inheritdoc/>
+    public override async Task<OrchestrationMetadata> WaitForInstanceCompletionAsync(
+        string instanceId, bool getInputsAndOutputs = false, CancellationToken cancellation = default)
+    {
+        Check.NotNullOrEmpty(instanceId);
+        OrchestrationState state = await this.Client.WaitForOrchestrationAsync(
+            instanceId, null, TimeSpan.MaxValue, cancellation);
+        return this.ToMetadata(state, getInputsAndOutputs);
+    }
+
+    /// <inheritdoc/>
+    public override async Task<OrchestrationMetadata> WaitForInstanceStartAsync(
+        string instanceId, bool getInputsAndOutputs = false, CancellationToken cancellation = default)
+    {
+        Check.NotNullOrEmpty(instanceId);
+
+        while (true)
+        {
+            OrchestrationMetadata? metadata = await this.GetInstanceMetadataAsync(
+                instanceId, getInputsAndOutputs, cancellation);
+            if (metadata is null)
+            {
+                throw new InvalidOperationException($"Orchestration with instanceId '{instanceId}' does not exist");
+            }
+
+            if (metadata.RuntimeStatus != OrchestrationRuntimeStatus.Pending)
+            {
+                // TODO: Evaluate what to do with "Suspended" state. Do we wait on that?
+                return metadata;
+            }
+
+            await Task.Delay(TimeSpan.FromSeconds(5), cancellation);
```

Comment:
5 seconds is pretty long. Was this chosen arbitrarily or did you copy it from somewhere else? I wonder if 1 second would be better...

Reply:
Changed to 1s. I think I used 5s as there was an instance of a 5s delay in the gRPC processor, but that is for another use case anyways.